tfhe/src/integer/server_key/radix_parallel/scalar_bitwise_op.rs

```diff
@@ -0,0 +1,342 @@
+use crate::integer::block_decomposition::{BlockDecomposer, DecomposableInto};
+use crate::integer::ciphertext::RadixCiphertext;
+use crate::integer::ServerKey;
+use rayon::prelude::*;
+
+impl ServerKey {
+    pub fn unchecked_scalar_bitand_parallelized<T>(
+        &self,
+        lhs: &RadixCiphertext,
+        rhs: T,
+    ) -> RadixCiphertext
+    where
+        T: DecomposableInto<u8>,
+    {
+        let mut result = lhs.clone();
+        self.unchecked_scalar_bitand_assign_parallelized(&mut result, rhs);
+        result
+    }
+
+    pub fn unchecked_scalar_bitand_assign_parallelized<T>(&self, lhs: &mut RadixCiphertext, rhs: T)
+    where
+        T: DecomposableInto<u8>,
+    {
+        let message_modulus = self.key.message_modulus.0;
+        let carry_modulus = self.key.carry_modulus.0;
+        let total_modulus = carry_modulus * message_modulus;
+        assert!(total_modulus.is_power_of_two());
+        assert!(total_modulus <= 256);
```

Comment:
The power of two is required as the block decomposer uses bitshifts and so the modulus must be a power of two.

The total_modulus is leftover from when I first started the implementation and tried to pack ciphertext but that was kinda useless as it meant we would need to unpack.

A more correct assert would be `message_modulus <= 256`, since we cast to u8.

Reply:
do all functions using the decomposer have the checks in place ? And noted for the message modulus, though for integer we could use large scalars right ? or are we limiting to u8 ?

Reply:
I don't think all functions that uses assert have the power of two assert

The assert on the modulus could be moved into the iter_as of the block decomposer so that it checks every use

Iter_as is generic, here (and every else where in shortint) we use u8 as its the scalar type used, and we don't give params that go beyond 8 bits of plaintext

Reply:
alright, I let you be the judge on the best place to have the assert and avoid bad surprises

Reply:
I kept the power of two assert , and moved the modulus one in the iter_as

Reply:
only the message modulus needs to be a power of 2 in the end ? no need for the carry to be a power of 2 as well ?

Reply:
ok I think I see why with the ilog2 call

Reply:
yes that because of the ilog2 call is only on the message modulus, not (mesage_modulus * carry_modulus) as I first tried